docs/design_proposals/new-render-and-deploy.md

```diff
@@ -0,0 +1,374 @@
+
+![Command Life Cycle](images/cmd-lifecycle.png)
+
+### New "render"
+
+Add a new render (has decided to name to `.manifests`, the following graphs still using ".render") in the `skaffold.yaml`. This section contains two modes to hydrate configuration: One is to point to a declarative Kptfile file via `.manifests.kpt` (a.k.a kpt-managed), the other is to write the hydration pipeline directly in the `skaffold.yaml` (a.k.a skaffold-managed).
+
+#### kpt-managed mode
+
+The kpt-managed mode is for customers who are already the `kpt` hydration users (using the "pipeline" in Kptfile) and understand [declarative application management](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/declarative-application-management.md). Thus, there’s no need to ask them to re-write the similar config in skaffold.yaml.
+
+
+![kpt-managed mode config](images/kpt-managed-mode.png)
+
+#### skaffold-managed mode
+
+![skaffold-managed mode config](images/new-workflow.png)
+
+The skaffold-managed hydration has three types of operations: `generate`, `transform` and `validate`. The dry config is manipulated through each operation in the order as they are written in the `skaffold.yaml`.
+
+-  **generate**  defines the source of the configurations. It contains the raw manifests (kubernetes resources), `kustomize` resources (kustomization.yaml files and base/overlay resources)  and/or `helm` charts as the configuration source. Besides, It provides the option to fetch and update the `kpt` packages as resources.
+
+	*Note*: `helm` is only presented to users as "helm charts". Under the hood, the helm charts are managed by the `kpt`. See kpt [helm-inflator](https://github.com/GoogleContainerTools/kpt-functions-catalog/tree/master/examples/contrib/helm-inflator) example (the image prvoides minimum helm features, more work is needed before it can be used in skaffold). This allows `skaffold` to unify different config resources and pass them all through the kpt validation pipeline. What’s more, users are no longer required to install `kustomize`, `helm` or `kpt` and to fix the version conflicts.
+
+- **transform** defines the operations to mutate the configuration. Different than the `kpt` transformers, `skaffold` only accepts a **whitelisted set** of transformers. This is to control the `skaffold dev` qualities and gives a reliable render experience.
+
+- **validate** defines the operations to validate the configuration. Same as transform, only a whitelisted set of validators are accepted. E.g. `kubeval`, `conftest`.
+	
+
+### "render" interface in `skaffold.yaml`
+
+``` yaml
+
+apiVersion: skaffold/v3
+kind: Config
+metadata:
+   name: skaffold-config
+build:
+
+render:
+   # Optional. kpt-managed mode. The path to the Kptfile. If provided, the other "render" options are not allowed.
+   kpt: [string]
+
+   # This field defines the dry manifests from a variety of sources.
+   generate:
```

Comment:
Will the default for `skaffold fix` once this change goes in be to treat the deploy yaml files provided as DRY config?  For example, once this change is in will skaffold's simple examples/ skaffold.yaml files all have build, render and deploy phases?  
Would this yaml be treated as dry config?
https://github.com/GoogleContainerTools/skaffold/blob/master/examples/microservices/leeroy-app/kubernetes/deployment.yaml

What would the `skaffold fix` yaml look like for:
examples/microservices `skaffold.yaml`
https://github.com/GoogleContainerTools/skaffold/blob/master/examples/microservices/skaffold.yaml

nit: might be helpful to add an example with a no-op render and how that is handled by `skaffold fix`

Reply:
Yes, I think those yaml files are dry config. 
for 
https://github.com/GoogleContainerTools/skaffold/blob/master/examples/microservices/skaffold.yaml, it would be like 

```yaml
build: NO CHANGE
render:
   generate:
     manifests:
     - leeroy-web/kubernetes
     - leeroy-app/kubernetes
deploy:
portForward: NO CHANGE
profiles: NO CHANGE
```